Formula/temporalite.rb

```diff
@@ -0,0 +1,18 @@
+class Temporalite < Formula
+  desc "Experimental distribution of Temporal that runs as a single process"
+  homepage "https://github.com/temporalio/temporalite"
+  url "https://github.com/temporalio/temporalite/archive/v0.3.0.tar.gz"
+  sha256 "a4980cca9b265694dca97e696fbe9136b2bb4ceecfc44a43cfc7d1ac55a3009b"
+  license "MIT"
+
+  depends_on "go" => :build
+
+  def install
+    system "go", "build", *std_go_args(ldflags: "-s -w"), "./cmd/temporalite"
+  end
+
+  test do
+    run_output = shell_output("#{bin}/temporalite --version")
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.